components/support/utils/src/main/java/mozilla/components/support/utils/Browsers.kt

```diff
@@ -346,10 +348,22 @@ class Browsers private constructor(
             } else {
                 PackageManager.MATCH_DEFAULT_ONLY
             }
-            return packageManager.queryIntentActivities(intent, flag)
-                    .orEmpty()
-                    .filter { it.activityInfo.exported && (includeThisApp ||
-                            it.activityInfo.packageName != context.packageName) }
+            @Suppress("TooGenericExceptionCaught")
+            // We need to catch most general exception because sometimes
+            // TransactionTooLargeException is not properly caught
+            val appList = try {
+                packageManager.queryIntentActivities(intent, flag)
+            } catch (e: Exception) {
+                if (e.cause is TransactionTooLargeException) {
+                    packageManager.queryIntentActivities(intent.apply {
+                        data = null
```

Comment:
Why are we searching again without `data`? Doesn't this find things we are not interested in?

Reply:
I am not sure it would... we do pass content type. The alternative would be returning just Fenix as a result.